compiler/rustc_resolve/src/check_unused.rs

```diff
@@ -182,6 +182,27 @@ impl<'a, 'b, 'tcx> Visitor<'a> for UnusedImportCheckVisitor<'a, 'b, 'tcx> {
             }
         } else {
             self.check_import(id);
+            let import = self.r.redundant_imports.remove(&id);
+            if let Some(import) = import {
+                if let ImportKind::Single {
+                    source,
+                    target,
+                    ref source_bindings,
+                    ref target_bindings,
+                    ..
+                } = import.kind
+                    && !self.unused_imports.contains_key(&self.base_id)
+                    && source == target
```

Comment:
What is this condition for?

Reply:
When I was running tests, I found that there would be some false positives for `use crate_name as new_crate_name`.I wanted to allow this.

Reply:
Could you give an example?
The logic in `check_for_redundant_imports` looks like it should work for any cases involving renaming too.

Reply:
> Could you give an example? The logic in `check_for_redundant_imports` looks like it should work for any cases involving renaming too.

Sorry, I made a mistake. This condition `source == target` is not necessary here.